impl/src/main/java/org/jboss/forge/roaster/spi/JavaParserImpl.java

```diff
@@ -55,13 +57,28 @@
       return parseUnit(data).getGoverningType();
    }
 
+   private Charset getDefaultEncoding()
+   {
+      return Roaster.getDefaultEncoding();
```

Comment:
That's the most drastic part of this refactor. To share the default encoding determination logic between JavaParserImpl.java and Roaster.java, the implementation of JavaParserImpl#getDefaultEncoding() is moved into the Roaster.java considering the dependency graph. I think that it's a  good design to share such a default(and minor) auto-detection logic for integrity and maintainability. If  it is a project rule that "JavaParserImpl should not have a dependency with Roaster", we must decide whether the Roaster#getDefaultEncoding() is moved into any utility class of the 'roaster-api' project or the implementation of logic is split into two project (roaster-api and jdt).